[webpack] Migrate most js tests

orangejenny opened the pull request:
## Technical Summary
This makes some updates to js testing infrastructure to support js tests that use webpack, and it migrates js tests except for reports and cloudcare.

I'm waiting on migrating reports & userreports tests until https://github.com/dimagi/commcare-hq/pull/35599 is live, just in case that one needs to be rolled back.

For cloudcare tests, I'll either incorporate them into https://github.com/dimagi/commcare-hq/pull/35594 or do them after that PR is released.

https://dimagi.atlassian.net/browse/SAAS-15985

## Safety Assurance

### Safety story
Primarily affects tests. There is one non-tests change (https://github.com/dimagi/commcare-hq/pull/35623/commits/1f43a0fb9eaaa3dfcf34002b1a27feb9506e5145). I've verified that translations still show up on webpack pages.

### Automated test coverage

yep

### QA Plan

nope

### Rollback instructions
- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

orangejenny committed:
Added mocha to alwaysIncludeApps

This is necessary because the mocha app itself doesn't have any templates that use webpack. It just has
mocha/base.html, which is only used as a parent for templates in other apps.

orangejenny committed:
Updated mocha/base.html to support webpack

Notably, in requirejs, tests do not include "base_main" js,
which includes a bunch of global UI-oriented code: analytics,
modals for accounting and for inactivity, etc. In webpack, these
are part of the "commcarehq" bundle, which can end up in the dependency
tree for tests - as an example, for hqwebapp, hqwebapps/js/widgets is a
dependency of tests, and it includes commcarehq. To deal with this, I'm
updating this template to not error out on this UI code. This is why the language script
is moved up (by moving it above webpack.html, it's now available to all of the
js in tests), and why the commcarehq-urls block is added (without this,
notifications_service, which is part of base_main, errors out because it depends
on querying the notifications url).

orangejenny committed:
Migrated hqwebapp tests to webpack

orangejenny committed:
Renamed exports template

orangejenny committed:
Switched webpack source map style from inline-source-map to eval-cheap-module-source-map

This is necessary to get proper line numbers to show up in mocha stack traces.

orangejenny committed:
Migrated locations tests to webpack

orangejenny committed:
Updated show_invalid_bootstrap3_files command to ignore mocha templates

Mocha templates are part of the django app they test, but they inherit from base.html
in the mocha django app, and they use B3 unless they'rr part of a split app that has
both B3 and B5 tests. See https://github.com/dimagi/commcare-hq/blob/98629eaea9bb2cd9984fc374eb25d3dd3a37d87a/corehq/apps/mocha/views.py#L13-L21

orangejenny committed:
Added managedPaths to webpack config

Silences warnings along the lines of "Managed item ... isn't a directory or doesn't contain a package.json"

orangejenny committed:
Updated webpack_bundles to generate bundles in tests if the manifest is present

orangejenny committed:
Moved webpack build before run_tests

orangejenny committed:
Removed hqwebapp/js/django from commcarehq.js

This module was implemented for requirejs and is not needed for webpack.

orangejenny committed:
Migrated case_importer tests to webpack

orangejenny committed:
Migrated exports tests to webpack

orangejenny committed:
Updated notifications tests to use hqDefine (ignore whitespace)

orangejenny committed:
Migrated notifications tests to webpack

orangejenny committed:
Updated generateDetails.js to skip test files when running in production mode

millerdev commented on corehq/apps/hqwebapp/templatetags/hq_shared_tags.py:
```diff
@@ -768,6 +765,11 @@ def webpack_bundles(entry_name):
             manifest = get_webpack_manifest('manifest_b3.json')
             webpack_folder = 'webpack_b3'
     except WebpackManifestNotFoundError:
+        # If we're in tests, the manifestly genuinely may not be available,
```
nit
```suggestion
        # If we're in tests, the manifest genuinely may not be available,
```

orangejenny committed:
Update corehq/apps/hqwebapp/templatetags/hq_shared_tags.py

Co-authored-by: Daniel Miller <dmiller@dimagi.com>

orangejenny committed:
Added test command

'yarn build' no longer includes dev dependencies, but 'yarn dev' uses the --watch flag so it never returns.

orangejenny committed:
Merge branch 'jls/mocha-experimental' of https://github.com/dimagi/commcare-hq into jls/mocha-experimental
